superset/migrations/shared/migrate_viz/processors.py

```diff
@@ -131,3 +133,59 @@ class MigrateSunburst(MigrateViz):
     source_viz_type = "sunburst"
     target_viz_type = "sunburst_v2"
     rename_keys = {"groupby": "columns"}
+
+
+class TimeseriesChart(MigrateViz):
+    has_x_axis_control = True
+
+    def _pre_action(self) -> None:
+        self.data["contributionMode"] = "row" if self.data.get("contribution") else None
+        self.data["zoomable"] = False if self.data.get("show_brush") == "no" else True
+        self.data["markerEnabled"] = self.data.get("show_markers") or False
+        self.data["y_axis_showminmax"] = True
+
+        bottom_margin = self.data.get("bottom_margin")
+        if self.data.get("x_axis_label") and (
+            not bottom_margin or bottom_margin == "auto"
+        ):
+            self.data["bottom_margin"] = 30
+
+        if (rolling_type := self.data.get("rolling_type")) and rolling_type != "None":
+            self.data["rolling_type"] = rolling_type
+
+        if time_compare := self.data.get("time_compare"):
+            self.data["time_compare"] = [
+                value + " ago" for value in as_list(time_compare) if value
```

Comment:
Are there causes when `value` is falsy?

Reply:
If I remember correctly, I added this check because old charts contained falsy values. Many of these safe checks were added as a result of trying to run this migration using our production database which contained really old charts.